### ISSUE TITLE
Intellisense for triggers, etc only works if on a new line after opening statement

### ISSUE THREAD

**Issue by brandonh-msft** [microsoft/vscode-azurelogicapps#46]
Looking at an empty logic app, one will see:
```json
{
    "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {},
    "triggers": {},
    "actions": {},
    "outputs": {}
}
```

If I then put my cursor here:
![image](https://user-images.githubusercontent.com/20270743/46560661-6a1b4380-c8a9-11e8-9357-49a855758441.png)

My intellisense will be unhelpful. Showing only past things I've typed in VSCode, or that VSCode sees in the current document:
![image](https://user-images.githubusercontent.com/20270743/46560692-90d97a00-c8a9-11e8-9883-5c934651f3fc.png)

However, if I insert a line break:
![image](https://user-images.githubusercontent.com/20270743/46560707-9d5dd280-c8a9-11e8-9bba-3081ae056d97.png)

I now get the expected autocompletes & snippets for the Logic Apps schema

**Comment by joechung-msft** [comment]
You must name the trigger first before being given suggestions:
![image](https://user-images.githubusercontent.com/26208574/46635589-d5635080-cb09-11e8-9f09-b23eb6795f68.png)

**Comment by brandonh-msft** [comment]
the issue is that the intellisense snippet for `recurrence` **does** show up, but only after a newline; giving it a name isn't required, as shown.

**Comment by joechung-msft** [comment]
Those are two different suggestions.  The one that shows up in the "triggers" property is for the "recurrence" snippet, while the one which shows up after you have entered a name for a trigger is for the "recurrence" property for a trigger.

**Comment by nzthiago** [comment]
It kind of all combines into making it more intuitive and easier to use, as per this and other ones we raised, we found it not very intuitive. If you have to name it and there's a template for different triggers, if the user goes to Triggers help by filling out a template with a place for them to name the trigger and then choose the trigger type

**Comment by joechung-msft** [comment]
If you're in the "triggers" property,, it should offer suggestions for snippets to fill in triggers for you.  The suggestions which it currently shows is an artifact of Code also suggesting previously entered strings in addition to snippets.

Once you have named a trigger, it should make suggestions based on it being a trigger.  

Once you enter two double quotes, the JSON is no longer valid so it will be difficult to make sound schema-based suggestions.

**Comment by joechung-msft** [comment]
When you have an empty object with no new line for "triggers", it should recommend snippets.  It is unclear why Code is not suggesting any snippets (maybe a Code limitation).

When you have an empty object with a new line for "triggers", it should and does recommend snippets (by design).

When you have entered a name for a trigger, e.g.,`"mytrigger":`, it should not recommend adding a snippet since that would result in invalid JSON.  It might be a limitation of Code that it does not suggest beginning an object here since the schema specifies that.

When you have entered `"mytrigger:"{}`, it should recommend trigger properties, but it should not recommend new snippets.  This is a limitation of the "snippet" approach in place since the snippets are JSON snippets only and do not understand the Logic Apps workflow definition schema.

**Comment by nzthiago** [comment]
Understand there are explanations for the current behavior. We are relaying our feedback on the experience and how confusing it seems to be unless you know what's needed to create the trigger or the rest of the workflow definition by heart. Hopefully, the LA team can work with the VS Code team to improve the end to end experience.

**Comment by joechung-msft** [comment]
I filed Microsoft/vscode#60704 for this issue.

**Comment by joechung-msft** [comment]
The VS Code team closed microsoft/vscode#60704 as "won't fix" so we will not be able to as well.
